src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift

```diff
@@ -456,46 +483,57 @@ extension RiskProvider: RiskProviding {
 			detectionMode: riskProvidingConfiguration.detectionMode
 		)
     }
-}
 
-extension RiskProvider: ActivityStateProviderDelegate {
+	private func updateActivityState(_ state: ActivityState) {
+		Log.info("RiskProvider: Update activity state to: \(state)", log: .riskDetection)
 
-	/// - NOTE: The activity state is propagated to subscribers
-	/// via a sink that can be found in the RiskProvider initializer.
-	func provideActivityState(_ state: ActivityState) {
-		activityState = state
-	}
+		self.activityState = state
 
-	private func _provideActivityState(_ state: ActivityState) {
 		targetQueue.async { [weak self] in
 			self?.consumers.forEach {
 				$0.didChangeActivityState?(state)
 			}
 		}
 	}
+
+	private func registerForPackageDownloadStatusUpdate() {
+		self.keyPackageDownload.statusDidChange = { [weak self] downloadStatus in
+			guard let self = self else { return }
+
+			switch downloadStatus {
+			case .downloading:
+				self.updateActivityState(.downloading)
+			default:
+				break
+			}
+		}
+	}
 }
 
 extension RiskProvider {
 	enum ActivityState {
 		case idle
+		case riskRequested
 		case downloading
 		case detecting
 
 		var isActive: Bool {
-			self != .idle
+			self == .downloading || self == .detecting
 		}
 	}
 }
 
 #if DEBUG
 extension RiskProvider {
 	private func _requestRiskLevel_Mock(userInitiated: Bool, completion: Completion? = nil) {
-		let risk = Risk.mocked
-
-		targetQueue.async {
-			completion?(.success(.mocked))
+		// Erase optionality.
+		let completion: Completion = { result in
+			completion?(result)
```

Comment:
Why leave the completion handler then optional?

Reply:
Why not? It can be used or not by the other parts of the app.
There is a scenario where you can request risk and the result comes from the consumer.

Reply:
Already bad enough to get results on two possible ways but how about turning the completion block of `_requestRiskLevel` and `calculateRiskLevel` into optionals then? Also, the `_requestRiskLevel_Mock` has an optional completion handler. This looks inconsistent.

Suggestion at the corresponding place:
`private func _requestRiskLevel(userInitiated: Bool, ignoreCachedSummary: Bool, completion: Completion? = nil)`

Reply:
Getting results over 2 possible ways is not something I have added in this PR. It was already there and I also don't like it. Needs to be refactored.

"but how about turning the completion block of _requestRiskLevel and calculateRiskLevel into optionals then?"
Now we are getting to the point. Thats exactly what I want to prevent with the wrapped completion.

I don't think, that wrapping the completion to erase optionality is a "bad" thing.
But I can understand, that its kind of hard to understand.
I will change every completion to optional again.